cmd/planner/main_test.go

```diff
@@ -17,111 +17,204 @@ import (
+	}
 }
 
 // createFlavor sends a create flavor request to the server
-func createFlavor(t *testing.T, port, flavor, platform string, labels []string, priority, pressure int) int {
-	t.Helper()
+func (ctx *testContext) createFlavor(flavor, platform string, labels []string, priority int) int {
+	ctx.t.Helper()
 
 	body := map[string]any{
-		"platform":         platform,
-		"labels":           labels,
-		"priority":         priority,
-		"minimum_pressure": pressure,
+		"platform": platform,
+		"labels":   labels,
+		"priority": priority,
 	}
 
 	b, err := json.Marshal(body)
-	if err != nil {
-		t.Fatalf("marshal payload: %v", err)
-	}
+	require.NoError(ctx.t, err, "marshal payload")
 
-	url := "http://localhost:" + port + "/api/v1/flavors/" + flavor
+	url := "http://localhost:" + ctx.port + "/api/v1/flavors/" + flavor
 
 	resp, err := http.Post(url, "application/json", bytes.NewReader(b))
-	if err != nil {
-		t.Fatalf("create flavor request: %v", err)
-	}
+	require.NoError(ctx.t, err, "create flavor request")
 	defer resp.Body.Close()
 
 	return resp.StatusCode
 }
 
 // getFlavorPressure sends a get flavor pressure request to the server
-func getFlavorPressure(t *testing.T, port, flavor string) map[string]int {
-	t.Helper()
+func (ctx *testContext) getFlavorPressure(flavor string) map[string]int {
+	ctx.t.Helper()
 
-	url := "http://localhost:" + port + "/api/v1/flavors/" + flavor + "/pressure"
+	url := "http://localhost:" + ctx.port + "/api/v1/flavors/" + flavor + "/pressure"
 
 	resp, err := http.Get(url)
-	if err != nil {
-		t.Fatalf("get flavor pressure request: %v", err)
-	}
+	require.NoError(ctx.t, err, "get flavor pressure request")
 	defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK {
-		t.Fatalf("unexpected status getting pressure: %d", resp.StatusCode)
-	}
+	require.Equal(ctx.t, http.StatusOK, resp.StatusCode, "unexpected status getting pressure")
 
 	var pressures map[string]int
-	if err := json.NewDecoder(resp.Body).Decode(&pressures); err != nil {
-		t.Fatalf("decode response: %v", err)
-	}
+	require.NoError(ctx.t, json.NewDecoder(resp.Body).Decode(&pressures), "decode response")
 
 	return pressures
 }
 
 // assertFlavorPressureEquals checks that the pressures map contains the expected pressure for the given flavor
-func assertFlavorPressureEquals(t *testing.T, pressures map[string]int, flavor string, expected int) {
-	t.Helper()
+func (ctx *testContext) assertFlavorPressureEquals(pressures map[string]int, flavor string, expected int) {
+	ctx.t.Helper()
 	value, exists := pressures[flavor]
-	if !exists {
-		t.Fatalf("expected flavor %q in response, got %+v", flavor, pressures)
-	}
-
-	if value != expected {
-		t.Fatalf("expected pressure %d for flavor %q, got %d", expected, flavor, value)
-	}
+	require.True(ctx.t, exists, "expected flavor %q in response, got %+v", flavor, pressures)
+	require.Equal(ctx.t, expected, value, "expected pressure %d for flavor %q, got %d", expected, flavor, value)
```

Comment:
I think this should be
```suggestion
	assert.True(ctx.t, exists, "expected flavor %q in response, got %+v", flavor, pressures)
	assert.Equal(ctx.t, expected, value, "expected pressure %d for flavor %q, got %d", expected, flavor, value)
```